### DISCUSSION TITLE
Odd issue with read-file-name and vertico

### DISCUSSION THREAD

**Discussion by artelse** [minad/vertico#484] [Q&A] [answered]
I'm having an odd issue with the code below. When `vertico` is enabled, the `read-file-name` function fails to return the new edited value. Vertico is standard in its settings, only `vertico-resize` and `vertico-cycle` are enabled. Running on a recent 30 master branch.
```
(defun ae/rename-current-buffer-file ()
  "Rename the current buffer's file, prompting for a new name."
  (interactive)
  (let ((old-name (buffer-file-name)))
    (when old-name
      (let* ((old-dir (file-name-directory old-name))
             (new-name (read-file-name "New name: " old-dir nil nil (file-name-nondirectory old-name))))
        (when (not (equal old-name new-name))
          (rename-file old-name new-name 1)
					(set-visited-file-name new-name)
          (message "File renamed to %s" new-name))
       (message "File-is-new? %s" new-name)))))
```

**Comment by artelse** [comment]
What is the difference between an issue (bug) and a discussion in general and in this case?

**Reply by minad** [reply, marked as answer]
It seems to be that there is no bug here. You can exit with the edited input by navigating to the prompt or pressing M-RET. This behavior is documented in the README, see https://github.com/minad/vertico?tab=readme-ov-file#key-bindings. Does this help?

**Reply by artelse** [reply]
Thanks @minad, I understand now why that happened. I think I will rewrite the function to just ask for the new title and rename the file accordingly or something similar.
